### PR TITLE
[processing] restore import into spatialite & spatialite execute sql algorithms

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , review appreciated.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
The executive sql alg looks good. The import alg is correctly ported, but when I ported the postgis import alg I also added some improvements which would be good to have here. Specifically, manually adding features to the exporter instead of calling the static export method allows cancelation and progress support, and allows use of the more flexible sources instead of vector layers (i.e. allowing import of only selected features, plus other future source only improvements)

**Comment by nyalldawson** [comment]
Ignore the bad autocorrections above! By the way, this is much appreciated!

File: python/plugins/processing/algs/qgis/SpatialiteExecuteSQL.py [hunk 1]
```diff
@@ -56,16 +56,19 @@ def displayName(self):
...
+        databaseuri = database.dataProvider().dataSourceUri()
+        uri = QgsDataSourceUri(databaseuri)
         if uri.database() is '':
-            if '|layerid' in database:
-                database = database[:database.find('|layerid')]
-            uri = QgsDataSourceUri('dbname=\'%s\'' % (database))
+            if '|layerid' in databaseuri:
+                databaseuri = databaseuri[:databaseuri.find('|layerid')]
+            uri = QgsDataSourceUri('dbname=\'%s\'' % (databaseuri))
         self.db = spatialite.GeoDB(uri)
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/SpatialiteExecuteSQL.py]: This should be just 'db', not self.db. In the new api algorithm execution should not alter the state of the algorithm instance itself.

**Comment by nirvn** [comment]
@nyalldawson , requested changes pushed into the latest commit. This progress support for the import algorithm is brilliant, thanks for pointing this out (and making it possible to do to begin with 😄).

**Pull request merged**